tools/migration/internal/logger.go

```diff
@@ -0,0 +1,46 @@
+package internal
+
+import (
+	"io"
+	"log"
+	"os"
+)
+
+// Logger logs migration events to disk and, if initialized with verbose,
+// stdout too.
+type Logger struct {
+	logFile *os.File
+}
+
+// NewLogger creates a new Logger.  All log writes go to f, the logging file.
+// If the verbose flag is true all log writes also go to stdout.
+func NewLogger(f *os.File, verbose bool) *Logger {
+	// by default just write to file
+	var w io.Writer
+	w = f
+	if verbose {
+		w = io.MultiWriter(f, os.Stdout)
+	}
+	log.SetOutput(w)
```

Comment:
Blocking: this global side-effect is out of place here (or possibly anywhere). Constructing an object should not change process-wide logging behaviour.

I thought that the point of this object was that migrations would log to it directly, bypassing the `log` package. In that case, the methods below should write to logFile directly.

If we think it's also practically necessary to set the global log output to this stream, so that code using the `log` package gets the same behaviour, then that call should be made in the `main` entrypoint or similar. For that, add an `Install()` or similar method to Logger that can be invoked when wanted.

Reply:
good point about setting global log.

Reply:
Good point.  I will get rid of the global set entirely.